packages/headless/src/controllers/relevance-inspector/headless-relevance-inspector.ts

```diff
@@ -197,27 +233,47 @@ export function buildRelevanceInspector(
         },
         userIdentities,
         rankingExpressions,
-        // fields: [], TODO: allow to fetch fields from the API
+        fieldsDescription,
+        fetchAllFields,
       };
     },
 
     enable() {
       dispatch(enableDebug());
+      warnProductionEnvironment('debug');
     },
 
     disable() {
       dispatch(disableDebug());
+      dispatch(disableFetchAllFields());
+    },
+
+    enableFetchAllFields() {
+      dispatch(enableFetchAllFields());
+      warnProductionEnvironment('fetchAllFields');
+    },
+
+    disableFetchAllFields() {
+      dispatch(disableFetchAllFields());
+    },
+
+    fetchFieldsDescription() {
```

Comment:
I see you resolved my suggestion on placing field functionality in a dedicated `FieldInspector` controller, but there was no explanation or change.

Why do you think placing the new functionality on the relevance inspector is the better design?

Reply:
I see them as very similar things at the end of they day: "Tooling inside headless that allows you to debug and build your app".

Seeing the debug output/execution from the API, being able to retrieve ranking information for results, and being able to retrieve all metadata from your index are all related to the development phase, and not things you would show to your end users.

So IMO it's cleaner to have everything closely related to the same task be part of the same "tool box" (the relevance inspector controller).

Reply:
Ok, I saw `fetchFieldsDescription` less to do with debugging and relevance, since it does not change how the sapi `v2/` call behaves.

I saw it as a separate concern, to help explore fields, and perhaps one day filter by facet kind, sort etc., or have these available in state as properties. A dedicated controller could auto-fetch the data when instantiated for example, 

Another way to look at it is, multiple use-cases might want to offer an ability to explore fields. However, `debug` and `debugFields` are specific to the search api, and might not be supported by, or relevant to, the other apis.

Or is this over-engineering, anticipating? We can also extract to a dedicated controller later if needed. In this case, either the way to explore fields will be unique for the `search` use-case compared to others, or we would export the new controller for the `search` use-case and deprecate the method on the `RelevanceInspector` controller.

Reply:
If different use case (commerce) need to also have the mechanism to `fetch fields description`, then the actions are present and should be easy to re-use in a new (commerce) related debugging tool if we need it.

I would expect that a "debugging tool for a commerce listing page" would need different thing than a "debugging tool for a classic search page". Maybe the commerce tool would also want to list fields description in some manner, and so just using the action should do the trick ? 🤔

I think this is a bit over-anticipating as you say. Too much guessing about what is going to be needed...